### ISSUE TITLE
Support (load "filename")

### ISSUE THREAD

**Issue by jcburley** [candid82/joker#467]
Clojure supports `(load "filename")`, while Joker supports `(load 'namespace)` (which Clojure does not).

Seeing as Clojure's own `core.clj` uses `(load "core_otherstuff.clj")` to pull in definitions from "helper" files in the same directory as `core.clj` itself, it seems like this might be useful for Joker to support as well.

I've tried using `(load-file "filename")`, but that doesn't find `filename` relative to the namespace-defining directory, which is what Clojure's `load` does.

**Comment by jcburley** [comment]
Note that I'm not a big fan of overloading the argument(s) to `load` such that either a string _or_ a symbol name (rather than just the latter) is supported.

But maybe that's okay here, or perhaps a `load-relative` function could be created that does what Clojure's `load` does, or maybe Joker's current `load` behavior could be cloned into a new function (named `load-ns` or similar) and that current behavior removed from `load` after some time or set of releases.

**Comment by candid82** [comment]
I'd be hesitant to change current `load` behavior. Maybe we could add a 2-args arity to `load-file` that takes `opts` map as the second argument, with a key like `relative-to-ns` or something...

**Comment by jcburley** [comment]
> I'd be hesitant to change current `load` behavior. Maybe we could add a 2-args arity to `load-file` that takes `opts` map as the second argument, with a key like `relative-to-ns` or something...

To be clear, I'm suggesting only augmenting (via overloading), not changing any existing usage of, `load`, such that if it gets a string argument (which is currently not supported at all), it behaves more like Clojure; otherwise it'd handle a Symbol just as it currently does.

But I feel like I have very little grasp of the relevant history or thinking around these areas. E.g. is Clojure's `load` considered an archaic kludge? Why did Joker implement `load` to take a Symbol, rather than a String, and otherwise behave so differently? Is the sort of "side loading" that's done by `core.clj` frowned upon for some reason?

Since I don't have a terribly strong need for the functionality (other than I'm toying with implementing something in `core.joke` that is side-loaded via `core.clj`), I don't have a strong opinion on how it's done, other than I tend to default to "be as much like Clojure as feasible".

**Comment by candid82** [comment]
I don't quite remember why `load` was implemented to take a namespace name (a symbol) rather than a resource name (note that Clojure's version doesn't actually take a filename, it takes a "resource" name). Perhaps because I did not want to introduce the concept of "resource" in Joker. 
Regardless, this functionality is not really useful for end users. It could potentially be used by Joker's internals, but until that need arises let's punt on it.
